### PR TITLE
Lwm2m with security

### PR DESCRIPTION
**Pull request by beingStrongeryqqq**
**What this PR does / why we need it**:
lwM2M e2e test with security.
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [x] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (rhoninl) [inline examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh:1]: remove if not used

**Review by tomqin93 — APPROVED** [review]

File: azure-pipelines/azure-pipelines.yml [hunk 2]
```diff
@@ -211,7 +211,7 @@ stages:
               kubectl delete -f $(System.DefaultWorkingDirectory)/examples/deviceshifu/demo_device/edgedevice-opcua/mock-device
             displayName: "Shifu demo device OPCUA E2E test"
 
-      - job: kind_e2e_test_lwm2m_without_security
+      - job: kind_e2e_test_socket
```
Comment (tomqin93) [inline azure-pipelines/azure-pipelines.yml:214]: Let's switch the test order of socket and lwm2m test for better readability

File: examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh [hunk 3]
```diff
@@ -5,6 +5,7 @@ writeData=88.8
 
 # Get the pod name of deviceshifu
 pod_name=$(kubectl get pods -n deviceshifu -l app=deviceshifu-lwm2m-deployment -o jsonpath='{.items[0].metadata.name}')
+service_name=$(kubectl get services -n deviceshifu -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh:8]: you should use `-l app=deviceshifu-lwm2m-deployment` to select the service

File: examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh [hunk 4]
```diff
@@ -27,22 +28,26 @@ for i in {1..5}; do
     echo "Received value: $out"
     
     # Check if the server response indicates an error
-    if [[ $out != "Error on reading object" ]]; then
+    if [[ $out != "Error on reading object" && -n "$out" ]]; then
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh:31]: check `-n "$out"` first

File: examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh [hunk 5]
```diff
@@ -27,22 +28,26 @@ for i in {1..5}; do
...
         break
     fi
     
-    echo "Device is unhealthy. Attempting to reconnect... ($i/5)"
+    echo "Device is unhealthy. Attempting to reconnect... ($i/10)"
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh:35]: revert to 5

File: examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh [hunk 6]
```diff
@@ -27,22 +28,26 @@ for i in {1..5}; do
...
     sleep 3
 done
 
-if [[ $out == "Error on reading object" ]]; then
-    echo "Device is still unhealthy after 5 attempts. Exiting..."
+if [[ $out == "Error on reading object" || -z "$out" ]]; then
+    echo "Device is still unhealthy after 10 attempts. Exiting..."
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh:40]: ditto

File: examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh [hunk 7]
```diff
@@ -27,22 +28,26 @@ for i in {1..5}; do
...
     fi
     
-    echo "Device is unhealthy. Attempting to reconnect... ($i/5)"
+    echo "Device is unhealthy. Attempting to reconnect... ($i/10)"
     sleep 3
 done
 
-if [[ $out == "Error on reading object" ]]; then
-    echo "Device is still unhealthy after 5 attempts. Exiting..."
+if [[ $out == "Error on reading object" || -z "$out" ]]; then
```
Comment (tomqin93) [inline examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh:39]: ditto

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@beingStrongeryqqq added some comments, the rest looks good

**Review by tomqin93 — APPROVED** [review]
LGTM

File: azure-pipelines/azure-pipelines.yml [hunk 8]
```diff
@@ -239,14 +239,23 @@ stages:
...
+          - script: |
+              set -e 
+              kubectl apply -f $(System.DefaultWorkingDirectory)/examples/lwM2MDeviceshifuWithoutSecurity/lwM2M
+              kubectl wait --for=condition=Available deploy/deviceshifu-lwm2m-deployment -n deviceshifu --timeout=150s
               kubectl apply -f $(System.DefaultWorkingDirectory)/examples/lwM2MDeviceshifuWithoutSecurity/mockdevice
               kubectl wait --for=condition=available --timeout=150s deployment/leshan-client -n deviceshifu
               bash $(System.DefaultWorkingDirectory)/examples/deviceshifu/mockdevice/lwM2M/checkoutput.sh
               kubectl delete -R -f $(System.DefaultWorkingDirectory)/examples/lwM2MDeviceshifuWithoutSecurity
-            displayName: "Shifu demo device lwM2M E2E test"
+            displayName: "Shifu demo device lwM2M E2E test without security"
```
Comment (rhoninl) [inline azure-pipelines/azure-pipelines.yml:258]: with security

**Review by rhoninl — APPROVED** [review]
LGTM

**Pull request merged**